core/src/main/java/org/axonframework/commandhandling/gateway/GatewayProxyFactory.java

```diff
@@ -181,7 +191,7 @@ public GatewayProxyFactory(CommandBus commandBus, RetryScheduler retryScheduler,
             InvocationHandler dispatcher = new DispatchOnInvocationHandler(commandBus, retryScheduler,
                                                                            dispatchInterceptors, extractors,
                                                                            commandCallbacks, true);
-            if (!Future.class.equals(gatewayMethod.getReturnType())) {
+            if (!Future.class.isAssignableFrom(gatewayMethod.getReturnType())) {
```

Comment:
That seems to skip the block for cases where the return type is `Object`.. Should I do this very verbose way instead?

```
if( !(CompletableFuture.class.isAssignableFrom(gatewayMethod.getReturnType())
                    || Future.class.isAssignableFrom(gatewayMethod.getReturnType())
                    || CompletionStage.class.isAssignableFrom(gatewayMethod.getReturnType()))) {
```

It is long, but should work.